kubernetes/src/main/java/io/kubernetes/client/openapi/JSON.java

```diff
@@ -17,41 +17,36 @@
 import com.google.gson.JsonElement;
 import com.google.gson.JsonParseException;
 import com.google.gson.TypeAdapter;
+import com.google.gson.internal.bind.util.ISO8601Utils;
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
 import io.gsonfire.GsonFireBuilder;
-import io.kubernetes.client.gson.V1StatusPreProcessor;
-import io.kubernetes.client.openapi.models.V1Status;
+import io.kubernetes.client.openapi.models.*;
 import java.io.IOException;
 import java.io.StringReader;
 import java.lang.reflect.Type;
 import java.text.DateFormat;
 import java.text.ParseException;
+import java.text.ParsePosition;
+import java.time.LocalDate;
+import java.time.OffsetDateTime;
+import java.time.format.DateTimeFormatter;
 import java.util.Date;
-import java.util.Locale;
 import java.util.Map;
 import okio.ByteString;
-import org.joda.time.DateTime;
-import org.joda.time.LocalDate;
-import org.joda.time.format.DateTimeFormatter;
-import org.joda.time.format.DateTimeFormatterBuilder;
-import org.joda.time.format.ISODateTimeFormat;
 
 public class JSON {
   private Gson gson;
   private boolean isLenientOnJson = false;
   private DateTypeAdapter dateTypeAdapter = new DateTypeAdapter();
   private SqlDateTypeAdapter sqlDateTypeAdapter = new SqlDateTypeAdapter();
-  private DateTimeTypeAdapter dateTimeTypeAdapter = new DateTimeTypeAdapter();
+  private OffsetDateTimeTypeAdapter offsetDateTimeTypeAdapter = new OffsetDateTimeTypeAdapter();
   private LocalDateTypeAdapter localDateTypeAdapter = new LocalDateTypeAdapter();
   private ByteArrayAdapter byteArrayAdapter = new ByteArrayAdapter();
 
   public static GsonBuilder createGson() {
     GsonFireBuilder fireBuilder = new GsonFireBuilder();
-    GsonBuilder builder =
-        fireBuilder
-            .registerPreProcessor(V1Status.class, new V1StatusPreProcessor())
```

Comment:
@dsyer removal of this line leads to the e2e failure. let's get it back.

Reply:
did you update the openapi-ignore file before re-generating the project?

https://github.com/kubernetes-client/java/blob/53114b0c517c3b47f865e06d9f18ff6b55f30cc4/kubernetes/.openapi-generator-ignore#L10

Reply:
Ah, well spotted. I didn't see that one. It's a manual change to a generated source. Would be nice to automate it at some point.

Reply:
It looks like it didn't fix the e2e tests though. So more :shrug: .